### PR TITLE
Adds garlic, and tweaks bloodsuckers a bunch.

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Adds garlic, and a garlic necklace that protects against sucks, atleast until it is taken off, plus garlic has effects on vampires(the race) and bloodsuckers, both separate, cleans up a bit of the vars and unnecesary comments, vassal racks now have a hard limit on vassals that is tied to the bloodsucker level, makes bloodsucker nutrition status more related to blood level, and regeneration speed varies more with blood level, oh and bloodsuckers will eject prosthetics and will be counted towards final death, to prevent some exploits. Also using fortitude while buckled will unbuckle you. Also also made regenerative cores use one proc instead of copypasta, and made it not fully heal, only remove the stuns of the bloodsucker, works normally for non bloodsuckers however.

## Why It's Good For The Game
Lots of code quality improvements which remove the unneded clutter should help, plus a lot of these changes are badly needed to make playing bloodsuckers interesting hopefully, and feed needed a bit of a counter anyways, and why not have an appropiately themed thing thatll prevent bloodsuckers? 

## Changelog
:cl:
add: Adds garlic, a mutation of onions
add: You can now make garlic necklaces.
tweak: Tweaked hunger to be more the same as blood level for bloodsuckers
tweak: Bloodsuckers no longer get zero healing from regenerative cores, the core now heals their wounds but not their blood.
balance: Bloodsucker heal is now based a lot more on blood level.
fix: You can no longer be effectively immortal when fully auged as a bloodsucker.
fix: Regenerative cores will regain their old names when they are renewed, no more working decayed cores.
code: Removes a lot of unnecesiry clutter of comments and tries to make the vars more consistent for bloodsucker code.
code: Made the regenerative core use one proc instead of copypasta
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/garlic.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/obj/item/seeds/garlic
```
Comment (Trilbyspaceclone) [inline code/modules/hydroponics/grown/garlic.dm:1]: I dont see the mutation for this tbh did you forget it or am I blind?

Comment (Arturlang) [reply code/modules/hydroponics/grown/garlic.dm:1]: Uhhh, i might have missed it lemme see

**Comment by Arturlang** [comment]
travis is bork and quoting a nonexistant var

**Comment by KathrinBailey** [comment]
Once this is done I might add garlic bread pizza.
I love making garlic bread pizza!
Just whip up some bread dough, dice up some garlic, if you're feeling like it, grab a mortar and pestle and crush that garlic into a paste, add a little white onion for flavour and crush a bunch of chilli up too for a kick! A little mozzarella on top and bam, delicious! Or the classic route, no cheese but mix a bunch of olive oil in! <3

**Comment by Ludox235** [comment]
merge? maybe? mayhaps? Merge? Please???

**Comment by Arturlang** [comment]
Yeah, this thing is pretty much ready and was testmerged for awhile, with issues fixed during it

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -159,7 +159,7 @@
 		mess = TRUE
 		update_icon()
 		return FALSE
-	if(isvamp(clonemind)) //If the mind is a bloodsucker
+	if(AmBloodsucker(clonemind)) //If the mind is a bloodsucker
 		return FALSE
```
Comment (Ghommie) [inline code/game/machinery/cloning.dm]:
```suggestion
	if(AmBloodsucker(clonemind.current)) //If the mind is a bloodsucker
		return FALSE
```
Also move it under the `if(!QDELETED(clonemind.current))` check at line 142.

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -855,7 +855,7 @@
 /mob/living/carbon/proc/can_defib()
 	var/tlimit = DEFIB_TIME_LIMIT * 10
 	var/obj/item/organ/heart = getorgan(/obj/item/organ/heart)
-	if(suiciding || hellbound || HAS_TRAIT(src, TRAIT_HUSK))
+	if(suiciding || hellbound || HAS_TRAIT(src, TRAIT_HUSK) || AmBloodsucker())
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
	if(suiciding || hellbound || HAS_TRAIT(src, TRAIT_HUSK) || AmBloodsucker(src))
```

File: code/modules/antagonists/bloodsucker/bloodsucker_life.dm [hunk 4]
```diff
@@ -247,10 +257,11 @@
 		if(power.active && !power.can_use_in_torpor)
 			power.DeactivatePower()
 
-
 /datum/antagonist/bloodsucker/proc/Torpor_End()
 	owner.current.stat = SOFT_CRIT
-	owner.current.cure_fakedeath("bloodsucker") // Come after SOFT_CRIT or else it fails
+	owner.current.remove_status_effect(STATUS_EFFECT_UNCONSCIOUS)
+	owner.current.cure_fakedeath("bloodsucker")
```
Comment (Ghommie) [inline code/modules/antagonists/bloodsucker/bloodsucker_life.dm]: It doesn't even use fakedeath() to start with anymore.

File: code/datums/components/crafting/recipes/recipes_clothing.dm [hunk 5]
```diff
@@ -287,3 +287,12 @@
 				/obj/item/bedsheet/cosmos = 1)
 	time = 60
 	category = CAT_CLOTHING
+
+
+/datum/crafting_recipe/wintercoat_cosmic
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_clothing.dm]:
```suggestion
/datum/crafting_recipe/garlic_necklace
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/game/machinery/cloning.dm (hunk 2 above)
Comment (Arturlang) [reply code/game/machinery/cloning.dm:163]: Not gonna make it only run if qdeleted, but its now under it

File: code/modules/antagonists/bloodsucker/bloodsucker_life.dm (hunk 4 above)
Comment (Arturlang) [reply code/modules/antagonists/bloodsucker/bloodsucker_life.dm]: Removed it, tested and it didnt break anything

**Review by Ghommie — APPROVED** [review]

**Pull request merged**